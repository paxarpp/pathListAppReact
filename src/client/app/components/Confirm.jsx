import React, { Component } from "react";

import Button from "../components/Button";

export default class Confirm extends Component {
  handlerConf=(status) =>() => {
    const { handler } = this.props;
    handler(status)
  }
  render() {
    return (
      <div className="popUpConfirm">
      <p>Уверены ?</p>
        <Button handler={this.handlerConf(true)} styleButton="confirm">
          {String.fromCharCode(10003)}
        </Button>
        <Button handler={this.handlerConf(false)} styleButton="delit">
          {String.fromCharCode(10006)}
        </Button>
      </div>
    );
  }
}
