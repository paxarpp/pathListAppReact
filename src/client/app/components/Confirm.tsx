import React, { PureComponent } from 'react';
import Icon from './Icon';
import styled, { css } from 'styled-components';

interface IProps {
  handler: (p: boolean) => boolean;
}

class Confirm extends PureComponent <IProps>{

  handlerTrue = () => {
    const { handler } = this.props;
    handler(true);
  }
  handlerFalse = () => {
    const { handler } = this.props;
    handler(false);
  }
  public render() {
    return (
      <PopUpConfirm>
        <p>Уверены ?</p>
        <LeftIcon onClick={this.handlerTrue} color="green" name="Done" />
        <RightIcon onClick={this.handlerFalse} color="red" name="Clear" />
      </PopUpConfirm>
    );
  }
} 

const PopUpConfirm = styled.div`
  width: 140px;
  height: 60px;
  border-radius: 10px;
  box-shadow: 3px 3px 10px 1px grey;
  position: absolute;
  top: 45%;
  right: 30%;
  background-color: #fff;
  z-index: 999;
  > p {
    margin: 0 1rem;
  }
`;

const animHover = css`
  translate: opacity 0.3s;
  opacity: 0.7;
  :hover {
    opacity: 1;
  }
`;

const LeftIcon = styled(Icon)`
  margin-left: 10px;
  ${animHover};
`;

const RightIcon = styled(Icon)`
  margin-left: 70px;
  ${animHover};
`;

export default Confirm;
