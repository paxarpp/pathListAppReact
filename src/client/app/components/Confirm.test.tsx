import React from 'react';
import renderer from 'react-test-renderer';
import Confirm from './Confirm';

test('Component', () => {
  const component = renderer.create(<Confirm handler={() => false} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
