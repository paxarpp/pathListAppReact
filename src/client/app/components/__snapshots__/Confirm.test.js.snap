// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component 1`] = `
<div
  className="popUpConfirm"
>
  <p>
    Уверены ?
  </p>
  <a
    className="confirm"
    onClick={[Function]}
  >
    ✓
  </a>
  <a
    className="delit"
    onClick={[Function]}
  >
    ✖
  </a>
</div>
`;
